src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,1441 @@
+import {VisualizerExportModule} from '@/visualizers/VisualizerInterface'
+import {VisualizerDefault} from '@/visualizers/VisualizerDefault'
+import {ceil, floor, sqrt} from 'mathjs'
+
+const FIBONACCI_NUMBERS = [
+    0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597,
+    2584, 4181, 6765, 10946, 17711, 28657, 46368, 75025, 121393, 196418,
+    317811, 514229, 832040, 1346269, 2178309, 3524578, 5702887, 9227465,
+    14930352, 24157817, 39088169, 63245986, 102334155, 165580141, 267914296,
+    433494437, 701408733, 1134903170, 1836311903, 2971215073, 4807526976,
+    7778742049, 12586269025,
+]
+const LUCAS_NUMBERS = [
+    2, 1, 3, 4, 7, 11, 18, 29, 47, 76, 123, 199, 322, 521, 843, 1346, 2207,
+    3571, 5778, 9349, 15127, 24476, 39603, 64079, 103682, 167761, 271443,
+    439204, 710647, 1149851, 1860498, 3010349, 4870847, 7881196, 12752043,
+    20633239, 33385282, 54018521, 87403803, 141422324, 228826127, 370248451,
+    599074578, 969323029, 1568397607, 2537720636, 4106118243, 6643838879,
+    10749957122,
+]
+
+const BLACK = '#000000'
+const WHITE = '#ffffff'
+
+const RED = '#ff5733'
+const ORANGE = '#ff9d33'
+const YELLOW = '#e6ff33'
+const GREEN = '#14cd33'
+const BLUE = '#3388ff'
+const PURPLE = '#8814cd'
+const CYAN = '#00ffff'
+const MAGENTA = '#ff00ff'
+const VERDANT = '#9BBF30'
+const VIOLET = '#7f00ff'
+const MUSTARD = '#ffdb58'
+const GRAY = '#808080'
+
+const HIGHLIGHT = RED
+const SHADE13 = 'rgb(255, 255, 255)'
+const SHADE12 = 'rgb(240, 240, 240)'
+const SHADE11 = 'rgb(220, 220, 220)'
+const SHADE10 = 'rgb(200, 200, 200)'
+const SHADE9 = 'rgb(180, 180, 180)'
+const SHADE8 = 'rgb(160, 160, 160)'
+const SHADE7 = 'rgb(140, 140, 140)'
+const SHADE6 = 'rgb(120, 120, 120)'
+const SHADE5 = 'rgb(100, 100, 100)'
+const SHADE4 = 'rgb(080, 080, 080)'
+const SHADE3 = 'rgb(060, 060, 060)'
+const SHADE2 = 'rgb(040, 040, 040)'
+const SHADE1 = 'rgb(020, 020, 020)'
+const SHADE0 = 'rgb(000, 000, 000)'
+
+//Path variables
+let x = 0
+let y = 0
+let currentDirection = 'right'
+let numberToTurnAt = 0
```

Comment:
Why are directions string constants when most other similar collections of options here are enums? Wouldn't an enum for directions be more consistent?

Reply:
Converted to enum, thanks.